### PR TITLE
feat(enhancement): Custom scan sounds

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature**

## Summary
You can now set custom sounds for cargo and outfit scanners. They're played when you scan a ship, or when someone scans your ship. If there's no custom sound set, the default one is used, for two reasons: to prevent situations where someone forgets to add a sound and there's no audible warning that the player's ship is being scanned, and to provide some backward compatibility.

There are no data changes in this PR.

## Testing Done
I've done some in-flight testing with custom outfits.

## Wiki Update
https://github.com/endless-sky/endless-sky-wiki/pull/15

## Performance Impact
N/A

### CODE REVIEW

File: source/Ship.cpp [hunk 1]
```diff
@@ -1857,7 +1863,7 @@ int Ship::Scan(const PlayerInfo &player)
 
 	// Check if either scanner has finished scanning.
 	bool startedScanning = false;
-	bool activeScanning = false;
+	int activeScanning = 0;
```
Comment (nickshanks) [inline source/Ship.cpp:1866]:
```suggestion
	int activeScanning = ShipEvent::NONE;
```

**Comment by nickshanks** [comment]
All outfits that do cargo or outfit scanning will play their sounds on top of each other — it could get quite cacophonous, perhaps a limit? Also, there's no de-duping of sounds between the cargo and outfit std::maps.

File: source/Outfit.h [hunk 2]
```diff
@@ -143,6 +146,8 @@ class Outfit : public Weapon {
 	std::map<const Sound *, int> jumpSounds;
 	std::map<const Sound *, int> jumpInSounds;
 	std::map<const Sound *, int> jumpOutSounds;
+	std::map<const Sound *, int> cargoScanSounds;
+	std::map<const Sound *, int> outfitScanSounds;
```
Comment (nickshanks) [inline source/Outfit.h:150]: Is the int used for anything? If not, these can just be a std::vector

Comment (TomGoodIdea) [reply source/Outfit.h:150]: I think hyperdrive sounds are handled in a similar way. So, if the ints aren't required, they can be removed there too.

Comment (RisingLeaf) [reply source/Outfit.h:150]: just read the comment above

Comment (TomGoodIdea) [reply source/Outfit.h:150]: ~~yeah, but I think it may not really work like this~~

File: source/Ship.cpp [hunk 3]
```diff
@@ -1857,7 +1863,7 @@ int Ship::Scan(const PlayerInfo &player)
...
 	int result = 0;
```
Comment (TomGoodIdea) [inline source/Ship.cpp:1867]:
```suggestion
 	int result = ShipEvent::NONE;
```

File: source/Ship.cpp [hunk 4]
```diff
@@ -1857,8 +1863,8 @@ int Ship::Scan(const PlayerInfo &player)
 
 	// Check if either scanner has finished scanning.
 	bool startedScanning = false;
-	bool activeScanning = false;
-	int result = 0;
+	int activeScanning = ShipEvent::NONE;
+ 	int result = ShipEvent::NONE;
```
Comment (TomGoodIdea) [inline source/Ship.cpp:1867]:
```suggestion
	int result = ShipEvent::NONE;
```

**Comment by Saugia** [comment]
> All outfits that do cargo or outfit scanning will play their sounds on top of each other — it could get quite cacophonous, perhaps a limit?

That's not much of a heavy concern. It's currently the case with engines, afterburners, and, well, weapons. It wouldn't make sense we do so for scanners as well; typically you wouldn't have more than a few of the above in variance. At that rate, it would be less of a matter of a limit, and more-so making it so that you don't often run more than a few different scanners, through balance acts and acquisition, the same that is done for the other cases.

File: source/Outfit.h (hunk 2 above)
Comment (TomGoodIdea) [reply source/Outfit.h:150]: I can open a refactor to change these maps to some other containers once this PR is merged.

Comment (TomGoodIdea) [reply source/Outfit.h:150]: what do you think would be best to use here, an `std::set`?

**Pull request merged**